src/ray/rpc/raylet/raylet_server.h

```diff
@@ -0,0 +1,258 @@
+                                 SendReplyCallback send_reply_callback) = 0;
+  /// Handle a `PushError` request.
+  virtual void HandlePushErrorRequest(const PushErrorRequest &request,
+                                      PushErrorReply *reply,
+                                      SendReplyCallback send_reply_callback) = 0;
+  /// Handle a `PushProfileEvents` request.
+  virtual void HandlePushProfileEventsRequest(const PushProfileEventsRequest &request,
+                                              PushProfileEventsReply *reply,
+                                              SendReplyCallback send_reply_callback) = 0;
+  /// Handle a `FreeObjectsInStoreInObjectStore` request.
+  virtual void HandleFreeObjectsInStoreRequest(const FreeObjectsInStoreRequest &request,
+                                               FreeObjectsInStoreReply *reply,
+                                               SendReplyCallback send_reply_callback) = 0;
+  /// Handle a `PrepareActorCheckpoint` request.
+  virtual void HandlePrepareActorCheckpointRequest(
+      const PrepareActorCheckpointRequest &request, PrepareActorCheckpointReply *reply,
+      SendReplyCallback send_reply_callback) = 0;
+  /// Handle a `NotifyActorResumedFromCheckpoint` request.
+  virtual void HandleNotifyActorResumedFromCheckpointRequest(
+      const NotifyActorResumedFromCheckpointRequest &request,
+      NotifyActorResumedFromCheckpointReply *reply,
+      SendReplyCallback send_reply_callback) = 0;
+  /// Handle a `SetResource` request.
+  virtual void HandleSetResourceRequest(const SetResourceRequest &request,
+                                        SetResourceReply *reply,
+                                        SendReplyCallback send_reply_callback) = 0;
+  /// Handle a `SetResourceReply` request.
+  virtual void HandleHeartbeatRequest(const HeartbeatRequest &request,
+                                      HeartbeatReply *reply,
+                                      SendReplyCallback send_reply_callback) = 0;
+};
+
+/// The `GrpcService` for `RayletGrpcService`.
+class RayletGrpcService : public GrpcService {
+ public:
+  /// Construct a `RayletGrpcService`.
+  ///
+  /// \param[in] io_service Service used to handle incoming requests
+  /// \param[in] handler The service handler that actually handle the requests.
+  RayletGrpcService(boost::asio::io_service &io_service,
+                    RayletServiceHandler &service_handler)
+      : GrpcService(io_service), service_handler_(service_handler){};
+
+ protected:
+  grpc::Service &GetGrpcService() override { return service_; }
+
+  void InitServerCallFactories(
+      const std::unique_ptr<grpc::ServerCompletionQueue> &cq,
+      std::vector<std::pair<std::unique_ptr<ServerCallFactory>, int>>
+          *server_call_factories_and_concurrencies) override {
+    // Initialize the factory for `RegisterClient` requests.
+    std::unique_ptr<ServerCallFactory> register_client_call_factory(
```

Comment:
It would be great to have a macro for these patterns, but if it's too much it can wait for a future PR.

Reply:
Good advise, I think I should open a PR for it.